File: backend/app/models/roadmap.py
from dataclasses import dataclass, field
from datetime import date, datetime


@dataclass
class Roadmap:
    id: str
    user_id: str
    learning_path_id: str | None
    title: str
    total_weeks: int
    status: str
    started_at: datetime | None
    target_completion_date: date | None
    created_at: datetime


@dataclass
class RoadmapWeek:
    id: str
    roadmap_id: str
    week_number: int
    theme: str
    summary: str | None
    unlock_date: date | None
    status: str

File: frontend/src/types/roadmap.ts
export type RoadmapStatus = "not_started" | "in_progress" | "completed";
export type WeekStatus = "locked" | "active" | "completed";

export interface RoadmapWeek {
  id: string;
  roadmap_id: string;
  week_number: number;
  theme: string;
  summary: string | null;
  unlock_date: string | null;
  status: WeekStatus;
}

export interface Roadmap {
  id: string;
  user_id: string;
  title: string;
  total_weeks: number;
  status: RoadmapStatus;
  started_at: string | null;
  target_completion_date: string | null;
  created_at: string;
}
